jwst/tweakreg/astrometric_utils.py

```diff
@@ -81,8 +86,15 @@ def create_astrometric_catalog(input_models, catalog="GAIADR3", output="ref_cat.
     radius, fiducial = compute_radius(outwcs)
 
     # perform query for this field-of-view
-    ref_dict = get_catalog(fiducial[0], fiducial[1], sr=radius, catalog=catalog)
-    colnames = ('ra', 'dec', 'mag', 'objID')
+    epoch = (
+        epoch
+        if epoch is not None
+        else input_models[0].meta.observation.date+input_models[0].meta.observation.time
+    )
+    # keep only decimal point and digit characters
+    epoch = float("".join(c for c in str(epoch) if c == "." or c.isdigit()))
```

Comment:
JWST "DATE-OBS" has the form "2023-06-29" and "TIME-OBS" has the syntax "01:23:45.678". So with these operations `epoch` becomes a combined string like "2023-06-2901:23:45.678". Line 95 then turns that into a float value of 20230629012345.678, which is no where close to what we need. We need a floating-point year, something like 2023.4958...  (i.e. almost half way through 2023). So not only is the epoch value incorrect, but I'm guessing it's also probably causing the errors seen in the regression tests: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/769/testReport/jwst.regtest/test_nircam_align_to_gaia/

You need to call some function (I'll be there's something appropriate in astropy) to convert a properly formatted date+time string into a float year, where a properly formatted date+time string has a "T" separator between the date and time portions of the string, i.e. YYYY-MM-DDThh:mm:ss.ss